src/main/java/com/microsoft/sqlserver/jdbc/SharedTimer.java

```diff
@@ -77,11 +81,13 @@ public synchronized void removeRef() {
      * When the caller is finished with the SharedTimer it must be released via {@link#removeRef}
      */
     public static synchronized SharedTimer getTimer() {
```

Comment:
The `synchronized` on the static method is no longer required here as its being manually managed via the `lock` object.